### ISSUE TITLE
Send from Web UI?

### ISSUE THREAD

**Issue by STaRDoGG** [gotify/server#276]
So, I'm in the Gotify WebUI, and I want to send a message to one of the Clients in my list of Clients (or perhaps even "broadcast" the same message to all of them), how do I do this?

One simple example is I have a password that I created on my desktop, and want to send it over to my phone via a gotify msg.

**Comment by jmattheis** [comment]
Hey @STaRDoGG. 

Clients aren't supposed to send messages only applications are (See https://gotify.net/docs/). To send a message you've to create an application and then push a message. See https://gotify.net/docs/pushmsg

Still, it may be useful to send messages via the ui, so I'm changing this issue to a feature request.

**Comment by 4oo4** [comment]
I would like to see this too, but understand why it's currently not there. I know when I switched to this from Pushbullet, I had to get used to the difference between an application and a client, whereas with Pushbullet a device kinda acted like both. So not the most important thing to have but might help people switch from Pushbullet.

**Comment by STaRDoGG** [comment]
Hmmmm... I promise I'm not trying to be a snot, lol, but I can't:

Send any notifs/messages from the Web UI to 1 or more apps (or devices)
Can't send any notifs/messages to other apps/clients/devices from the official Android app
Can't view notifs/msgs as theyc ome in from any windows client other than loading a browser

I was hoping to 100% replace Pushbullet, as it's very unreliable, and the devs are out to lunch, plus I love the idea of a self hosted notif server, but it doesn't really seem that gotify is meant as a replacement, so to speak? Some very basic (imho) things aren't there like just passing notifications around to all of your clients in any direction at will (i.e. my Android phone, tablet, windows machine, pi, etc.)

Is any of this planned (other than the web ui feature request you added)?

**Comment by STaRDoGG** [comment]
> I know when I switched to this from Pushbullet, I had to get used to the difference between an application and a client, whereas with Pushbullet a device kinda acted like both.

I agree; it is a lil 'weird' to get used to I spose, lol. But what also adds to the confusion is even in the literal Android 'app' you can't create and send from the app to anything. It's kinda just 'dead' unless it receives something from elsewhere.

**Comment by pm-pm** [comment]
Why would my mother want to send something back when she gets the info that a package for her arrived? When i want to chat with her, i use a chat program.

**Comment by pm-pm** [comment]
@STaRDoGG
Stop telling what you can't do. You can't even make coffee with the app!

With the gotify server and the gotify app you can be very well informed automatically!

**Comment by jmattheis** [comment]
@STaRDoGG 

> I was hoping to 100% replace Pushbullet, as it's very unreliable, and the devs are out to lunch, plus I love the idea of a self hosted notif server, but it doesn't really seem that gotify is meant as a replacement, so to speak?

Is it marketed as replacement for Pushbullet? I don't think so.

> Some very basic (imho) things aren't there like just passing notifications around to all of your clients in any direction at will (i.e. my Android phone, tablet, windows machine, pi, etc.)

Different people have different needs and it is impossible to satisfy everyone. Originally, I've developed Gotify just for notifications when certain events occur on my server like a ssh login.
I'm just sharing a solution to exactly my problem/use-case not someone else's.

> Is any of this planned (other than the web ui feature request you added)?

> Can't send any notifs/messages to other apps/clients/devices from the official Android app

https://github.com/gotify/android/issues/72 is the feature request for adding send functionality to the android app.

> Can't view notifs/msgs as they come in from any windows client other than loading a browser

Is this a problem? There are clients in the making by the community f.ex. https://github.com/ajmcateer/GotifyDesktop

**Comment by STaRDoGG** [comment]
Ahh see? I knew my comment would still be taken as offense, even though I mentioned twice that I didn't mean it that way. I was just trying to understand why so much functionality seems 'missing' in it (at least so far). 

Agreed, you personally may not market it as "a pushbullet replacement", but it usually is placed in the same group by others, so it's easy to get confused. i.e. any software that allows sending notifications w/3rd party software, and some websites, like this:

https://alternativeto.net/software/pushbullet/

And feature-wise it can get a bit confusing even going on the wording on your website. For example:

![image](https://user-images.githubusercontent.com/1524526/75946442-c26dc080-5e62-11ea-8fe2-a7de28c1b034.png)


> ... server features a WebUI and functionality for:

> sending messages via a REST-API

Could easily be taken to mean you can send messages from the webui, that gets sent via a rest-api. 

And earlier you said:

> Clients aren't supposed to send messages only applications are

And your website says:

> managing users, **clients** and applications

(I know one says 'sending' and one says 'managing', but it'd be easy to assume managing also includes sending until you go to try it and find out that you can't.)

Anyway, again, not botching, even though I know it sounds that way. Just trying to dial in exactly what Gotify "does", and doesn't do, to see if it's something I need, or need to go back to something else instead. Though I'd love to see Gotify have the features I mention since I like it's looks, and self-hosting ability.

**Comment by jmattheis** [comment]
I'm not offended (:.

Yep, I think you always can take different meanings out of texts, but your interpretation makes sense. 

As a conclusion, sending messages from the ui/android app is a planned feature.

**Comment by jmattheis** [comment]
It is a feature request, therefore the issue should stay open.

**Comment by pm-pm** [comment]
Sending messages from web ui could be ok, but not from app???

I, as an family admin, have to mange "adressbooks", accessrights, etc. ?

If you enable "write from app", please make "a remote switch off" in the ui admin user panel.
So i can HIDE this point in the user app.

Thanks.

**Comment by STaRDoGG** [comment]
@pm-pm 

WebUI in Pushover:

![image](https://cms-assets.tutsplus.com/uploads/users/317/posts/22264/image/send-alert.jpg)

Pushover app:

![image](https://content.invisioncic.com/r229491/monthly_2016_09/alfred-pushover.png.6afe3ae9fff9c10ad63103e672df4b28.png)


These are pretty basic things; you don't have to manage addressbooks, etc. LOL (any more than you already do at least).

**Comment by pm-pm** [comment]
@STaRDoGG 

We are talking about "gotify". If you need something else please use something else!
I use this software BECAUSE it has NOT such features.

I do not want that userA can send messages to userB. They shout NOT know about each other!
Gotify is a SYSTEM-to-SYSTEM Message program, not a USER-to-USER Message program.

Please try to find something that suits you better.

**Comment by STaRDoGG** [comment]
@pm-pm How about "no?"

If you want it all to be one way, where they don't know about each other, just email them from a "no-reply@here.com" email, with BCC.

Sounds like that suits you better. Use it.

**Comment by pm-pm** [comment]
Please use the correct pills again. Thanks. ;-)

**Comment by STaRDoGG** [comment]
> Please use the correct pills again. Thanks. ;-)

@pm-pm Likewise .... ;)

**Comment by beac0n** [comment]
@jmattheis I'm interested in this feature as well, and I would like to implement it :-)

I see this button on the lower right in the web ui, which seemingly has no function (If I click on it, nothing happens.)
![image](https://user-images.githubusercontent.com/1389452/104240671-bd06dc80-545c-11eb-997e-5ad04687874b.png)

I would hide that functionality behind this button. Any objections/hints where to implement this in the frontend?

**Comment by ansonhorse** [comment]
> @jmattheis I'm interested in this feature as well, and I would like to implement it :-)
> 
> I see this button on the lower right in the web ui, which seemingly has no function (If I click on it, nothing happens.)
> ![image](https://user-images.githubusercontent.com/1389452/104240671-bd06dc80-545c-11eb-997e-5ad04687874b.png)
> 
> I would hide that functionality behind this button. Any objections/hints where to implement this in the frontend?

This is a `ScrollUpButton`, it helps scrolling to the top of the page.
[ui/src/common/ScrollUpButton.tsx](https://github.com/gotify/server/blob/master/ui/src/common/ScrollUpButton.tsx)
In my opinion, maybe it's better to hide it when there's no need to scroll to top.

**Comment by beac0n** [comment]
> This is a ScrollUpButton, it helps scrolling to the top of the page.
That makes sense :joy:. I thought it was smth like a Menu opener button. Instead of hiding, maybe a tooltip would be enough?

**Comment by jmattheis** [comment]
I'd be in favor of hiding the button and only show it after the user has scrolled down a bit.

**Comment by jmattheis** [comment]
It could be implemented as a dialog which will be opened via a button on the messages page of an application. 
![image](https://user-images.githubusercontent.com/14895212/104360734-6efbe280-5509-11eb-8865-0a98845b0b40.png)

**Comment by zivkovic** [comment]
I see this option exists in android app. Would be great to have it inside of Web UI also.

**Comment by AlliotTech** [comment]
Implemented on this PR: https://github.com/gotify/server/pull/897

**Comment by GitCourser** [comment]
> Implemented on this PR: [#897](https://github.com/gotify/server/pull/897)

I see that this merge has a “Push Message” button, but why can‘t I see it, the latest version v2.9.0

<img width="1426" height="203" alt="Image" src="https://github.com/user-attachments/assets/3793645e-ecb9-4a2f-8cfc-e3236ad8d0ef" />

**Comment by eternal-flame-AD** [comment]
You need to select an app first @GitCourser  , this button does not appear on the all messages page.
